src/Sylius/Bundle/WebBundle/Resources/config/routing/backend/product_archetype.yml

```diff
@@ -30,6 +30,24 @@ sylius_backend_product_archetype_update:
             template: SyliusWebBundle:Backend/ProductArchetype:update.html.twig
             redirect: sylius_backend_product_archetype_index
 
+sylius_backend_product_archetype_code_product_index:
+    path: /code/{code}/products
```

Comment:
The paths shouldn't contain * code* or *id* at the beginning. I think `/{code}/products` and `/{id}/products` would be enough. You can split them by adding [requirements](http://symfony.com/doc/current/book/routing.html#adding-requirements). Just remember, that first match will win!